File: Sorting/Sort an array in one swap whose two elements are swapped.cpp
#include<bits/stdc++.h>
using namespace std;

void OneSwap(int arr[],int n)
{
  int i=-1,j=-1;
  for(int k=1;k<n;++k)
  {
    if(arr[k-1]>arr[k])
    {
      if(i==-1)
      {
        i = k-1;
        j=k;//adjacent element swap
      }
      else
      {
        j = k; 
      }
    }
  }
  swap(arr[i],arr[j]);
}

int main()
{
  int arr[] = {3, 5, 7, 6, 8, 9};
  int n = sizeof(arr)/sizeof(arr[0]);
  OneSwap(arr,n);
  for(int i=0;i<n;++i) cout<<arr[i]<<" ";
  return 0;
}

File: Array/Sort an array in one swap whose two elements are swapped.py
def oneSwap(arr):
    a = -1
    b = -1
    for i in range(1,len(arr)):
        if arr[i-1] > arr[i] and a==-1:
            a = i-1
        else:
            b = i

    arr[a],arr[b] = arr[b],arr[a]
    return arr

arr = [3, 5, 6, 9, 8, 7]
arr = oneSwap(arr)
print(arr)
